### PR TITLE
GEP 10 - Container runtime extensions

### PR DESCRIPTION
**Pull request by nimrodoron**
**What this PR does / why we need it**:
GEP-10 implementation Creating new Container runtime extensions

Part of gardener/gardener#1454
✅ Depends on: https://github.com/gardener/gardener/pull/2035
❌Revendor

```noteworthy developer
Due to the introduction of the new [`ContainerRuntime` extensibility contract](https://github.com/gardener/gardener/blob/master/docs/extensions/containerruntime.md) a new generic controller + reconciler has been introduced. It allows developing an external container runtime extension controller that works on the `extensions.gardener.cloud/v1alpha1.ContainerRuntime` resource (similar to the other extension resources). Information for developing such an extension can be found [here](https://github.com/gardener/gardener/tree/master/docs/extensions/containerruntime.md).
```

### CODE REVIEW

File: pkg/controller/containerruntime/actuator.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"context"
+
+	extensioncontroller "github.com/gardener/gardener-extensions/pkg/controller"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+// Actuator acts upon Extension resources.
```
Comment (rfranzke) [inline pkg/controller/containerruntime/actuator.go]: rename `Extension` to `ContainerRuntime` everywhere here

File: pkg/controller/containerruntime/controller.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// FinalizerPrefix is the prefix name of the finalizer written by this controller.
+	FinalizerName = "extensions.gardener.cloud/container-runtime"
```
Comment (rfranzke) [inline pkg/controller/containerruntime/controller.go]:
```suggestion
	FinalizerName = "extensions.gardener.cloud/containerruntime"
```

**Comment by rfranzke** [comment]
@nimrodoron can you add a small release note for developers here as well, please?

**Comment by nimrodoron** [comment]
> @nimrodoron can you add a small release note for developers here as well, please?

Yes, done.

**Comment by rfranzke** [comment]
Hello @nimrodoron, gardener/gardener#2035 is merged now - could you revendor + rebase this PR so that we can merge it as well?

**Pull request closed without merging**